genie-core/src/main/java/com/netflix/genie/core/util/MetricsUtils.java

```diff
@@ -117,16 +113,30 @@ public static void addCommonJobWorkflowMetricTags(final JobExecutionEnvironment
      * @param cluster Cluster chosen for running the command
      * @param command Command chosen to be executed
      */
-    public static void addCommonJobWorkflowMetricTags(final Cluster cluster,
-                                                      final Command command,
-                                                      final Map<String, String> tags) {
+    public static void addClusterAndCommandTags(final Cluster cluster,
+                                                final Command command,
+                                                final Map<String, String> tags) {
         /* Add cluster tags */
-        tags.put(MetricsConstants.TagKeys.CLUSTER_NAME, cluster.getName());
-        tags.put(MetricsConstants.TagKeys.CLUSTER_ID, cluster.getId().orElse(NO_ID_FOUND));
+        tags.put(MetricsConstants.TagKeys.CLUSTER_NAME,
+            cluster != null
+                ? cluster.getName()
+                : MetricsConstants.TagValues.NO_CLUSTER_FOUND);
```

Comment:
Same for `MetricsConstants.TagKeys....` here and the closing `);` here and in the rest of this block